rfc/0033-quilt-mappings-and-clean-room.md

```diff
@@ -0,0 +1,97 @@
+
+## Motivation
+
+In order to understand the motivation for this RFC, it's important to look at how we got here. A lot has changed since Yarn was concieved in 2016, and the approach it takes isn't really relevant in today's modding landscape.
+
+### Background
+
+Yarn was originally created on August 12, 2016. It was created for a project called Open Mod Loader, the first iteration of what would be known as Fabric.
+
+It would remain in relative obscurity until October of 2018, when Yarn began tracking early snapshots for Minecraft 1.14.
+
+### Rift
+
+During the 1.13 development cycle, Forge chose to rewrite most of their scaffolding, and a stable Forge build didn't appear until Minecraft 1.14. Rift was created in response to this, as a lightweight mod loader that could do basic coremodding on Minecraft 1.13.
+
+However, Rift made use of parts of the Forge toolchain without permission. Forge declined to grant this permission to them after the fact, and Rift-related discussions were effectively banned in most modding-related communities.
+
+### Yarn's Motivation
+
+Due to the strict licensing of some parts of the Forge toolchain at the time, Fabric needed to avoid making use of Forge projects such as MCP, and thus avoid making the same mistakes that Rift did. In order to achieve this, Yarn was licensed under the [Creative Commons Zero License](https://creativecommons.org/share-your-work/public-domain/cc0/) (CC0), a public domain dedication.
+
+Making use of the CC0 in this way meant that all names in Yarn needed to be original contributions - it would have been unacceptable for a name to make it in from another mappings project, unless that project was also licensed under the CC0. For this reason, Yarn needed to take a cleanroom approach, rejecting contributions that added names from other mappings projects, and banning discussions of names from non-CC0 projects in related community spaces.
+
+### Issues With the Cleanroom
+
+While discussing how Quilt's approach to mappings should work, the Quilt Mappings Team has made the following observations:
+
+* Almost everyone that ended up taking up Fabric and Yarn were switching from other projects, generally Spigot or Forge
+* The Yarn project has never enforced a CLA, which means that there's no real certification from contributors that the names they've contributed are original
+* Some Yarn names are clearly inspired by or lifted from MCP, and couldn't have been named how they are without it
+* Some Yarn contributors have been known to be working with other mappings projects, either in their own mods or directly, while remaining active Yarn contributors - even with video evidence, in some cases
+
+For these reasons, the Quilt Mappings Team has concluded that it's impossible to say with absolute certainty that all Yarn names are unique, and not inspired by (or copied from) other mappings projects. This means that the CC0 license applied to the Yarn project is quite likely to be invalid.
```

Comment:
From section 4c of CC0:

> Affirmer disclaims responsibility for clearing rights of other persons that may apply to the Work or any use thereof, including without limitation any person's Copyright and Related Rights in the Work. Further, Affirmer disclaims responsibility for obtaining any necessary consents, permissions or other rights required for any use of the Work.

As discussed earlier, it seems more accurate to say it's unclear how much of the contents of the Yarn project are actually free of copyright claims, since the CC0 license is not invalidated by those competing copyright claims.

Reply:
I agree with zml here.
> This means that the CC0 license applied to the Yarn project is quite likely to be invalid.
I don't think this is right. CC0 doesn't claim that everything is original work.
In fact, as far as I understood, we're not even changing the license.
From the CC0 FAQ:
> The person applying CC0 to their work is not guaranteeing anything about it, including whether she owns the copyright or has cleared any uses of third-party content that her work may be based on or incorporate.
We're not planning to drop CC0 as far as I know, we're only dropping the (informal) claim that the mappings are original creations. And we're going a step further to explicitly accept non-fully original mappings and attributing them.

I think this sentence can simply be dropped from the RFC.